install/kubernetes/cilium/templates/hubble/tls-helm/ca-secret.yaml

```diff
@@ -0,0 +1,11 @@
+{{- if and .Values.hubble.enabled .Values.hubble.tls.enabled .Values.hubble.tls.auto.enabled (eq .Values.hubble.tls.auto.method "helm") }}
```

Comment:
This needs a check for `and (.Values.agent (not .Values.preflight.enabled))`. Otherwise, these secrets are created in the preflight check, which is wrong (as all other values are true here by default).

https://docs.cilium.io/en/v1.10/operations/upgrade/#running-pre-flight-check-required

You removed the `and (.Values.agent (not .Values.preflight.enabled))` check everywhere. I'm a bit nervous about that, we had issues in the past where the preflight check created artifacts that later broke the installation.

Reply:
Currently `and (.Values.agent (not .Values.preflight.enabled))` check is only apply to `tls-provided` and `tls-helm`.  
`tls-cronjob` is not using this check:
https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/hubble/generate-certs-cronjob.yaml#L1
And deployment is not using as well:
https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/hubble-relay/deployment.yaml#L1
Never the less, not all certs of `tls-provided` and `tls-helm` are using this check, named [`hubble-relay-client-certs`](https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/hubble/relay-tls-client-secret.yaml#L1), [`hubble-relay-server-certs`](https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/hubble/relay-tls-server-secret.yaml#L1) and [`hubble-ui-client-certs`](https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/hubble/ui-tls-client-secret.yaml#L1)

So, my point is this check is not universal, my opinion this remove this and let `hubble.enabled` take care the rest.
WDYT, @gandro?

Reply:
> So, my point is this check is not universal, my opinion this remove this and let hubble.enabled take care the rest.
WDYT, @gandro?

The problem is that with the values we have in the docs, https://docs.cilium.io/en/v1.10/operations/upgrade/#running-pre-flight-check-required, users will have `hubble.enabled=true`. So just checking for `hubble.enabled` will not work with the existing documentation.

If users follow the documentation as it is right now, to run the preflight check, they will have the following values:

```ini
preflight.enabled=true # explicitly set by the user
agent=false # explicitly set by the user
operator.enabled=false  # explicitly set by the user
hubble.enabled=true # default value
hubble.tls.auto.enabled=true # default value)
hubble.tls.auto.method="helm" # default value)
hubble.relay.enabled=true # default value)
```

This is why the `tls-helm` needs to check for `preflight.enabled` and `agent`, because all other values are default to true, so it will be applied (even though it shouldn't).

Basically the rule is: If a component is enabled by default, then it needs to check for `preflight.enabled` to disable itself in preflight mode. This is why `hubble-relay` or `tls-cronjob` currently do not have the preflight check, because they are not enabled by default.

But I fully agree that the inconsistency currently is not great. So let's fix it properly and as simple as possible.
My proposal would be the following.

- `hubble-server-certs` should only be created if `agent` is true.
   - Because it's only used by cilium-agent.
  -  In both modes: `tls-helm` and `tls-provided`.
  - The full check would be: `and .Values.agent .Values.hubble.enabled .Values.hubble.tls.enabled`
- `hubble-ca-secret` should only be created if `.agent` or  `hubble.relay.enabled` are true
  -  Because it's used by Agent or Relay, or UI (but UI requires Relay, so checking only for Relay is fine)
  -  In all three modes:`tls-helm`, `tls-provided`, `tls-cronjob`
  - The full check would be: `and (or .Values.agent .Values.hubble.relay.enabled) .Values.hubble.enabled .Values.hubble.tls.enabled`

The rest (`hubble-relay-server-certs`, `hubble-relay-client-certs`, `hubble-ui-client-certs`) already have to correct checks.

Reply:
Thank for your suggestion, it's updated.
I also checked `clustermesh` tls to match the logic, but I think you need to double check.